packages/shared_preferences/shared_preferences_linux/example/pubspec.yaml

```diff
@@ -0,0 +1,14 @@
+name: shared_preferences_example
+description: Demonstrates how to use the shared_preferences plugin.
+
+dependencies:
+  flutter:
+    sdk: flutter
+  shared_preferences:
+    path: ../../shared_preferences/
```

Comment:
This won't work if pulled via pub get. See https://github.com/flutter/plugins/blob/master/packages/path_provider/path_provider_linux/example/pubspec.yaml#L8 for an example dependency structure; the user-facing package should be depended on via a normal pub reference, and the _linux version should be an override (so it doesn't collide when the endorsement is added).

Reply:
I think I've got this right, but please check carefully.

Reply:
This is still failing solving on the bots:
```
ERR : Because every version of shared_preferences_linux from path depends on path_provider from hosted and all_plugins depends on path_provider from path, shared_preferences_linux from path is forbidden.

    | So, because all_plugins depends on shared_preferences_linux from path, version solving failed.
```

Looking back at [the series of PRs for path_provider](https://github.com/flutter/plugins/pulls?q=is%3Apr+path_provider+linux+is%3Aclosed+), it looks like the _linux package needed to depend on `any` for the main package: https://github.com/flutter/plugins/pull/2589/files#diff-8e0e3207e3fe6475f0fa024e1fcd9ffb

(Sorry I'm not able to offer clearer guidance, tangled package dependencies isn't something I have much previous experience with.)

Reply:
A bit of trial and error later and it seems to work now.